File: CS/WinWebSolution.Win/WinApplicationE1344.cs
using DevExpress.ExpressApp;
using WinWebSolution.Module;

namespace WinWebSolution.Win {
    partial class WinWebSolutionWindowsFormsApplication: ISupportChangeDatabaseAtRuntime {
	public void ChangeTo(string newConnectionString) {
	    ShowViewStrategy.CloseAllWindows();
	    IXpoDataStoreProvider dataStoreProvider = new ConnectionStringDataStoreProvider(newConnectionString);
	    Setup(new ExpressApplicationSetupParameters(ApplicationName, CreateDefaultObjectSpaceProvider(dataStoreProvider), ControllersManager, DetailViewItemsFactory, Model, Modules));
	    ShowViewStrategy.ShowStartupWindow();
	    if (SplashScreen != null) {
	        SplashScreen.Stop();
	    }
	}
    }
}

File: VB/WinWebSolution.Win/WinApplicationE1344.vb
Imports DevExpress.ExpressApp
Imports WinWebSolution.Module

Namespace WinWebSolution.Win
    Partial Public Class WinWebSolutionWindowsFormsApplication
        Implements ISupportChangeDatabaseAtRuntime
        Public Sub ChangeTo(ByVal newConnectionString As String) Implements ISupportChangeDatabaseAtRuntime.ChangeTo
            ShowViewStrategy.CloseAllWindows()
            Dim dataStoreProvider As IXpoDataStoreProvider = New ConnectionStringDataStoreProvider(newConnectionString)
            Setup(Me.ApplicationName, CreateDefaultObjectSpaceProvider(dataStoreProvider))
            ShowViewStrategy.ShowStartupWindow()
        End Sub
    End Class
End Namespace